sdk/core/azure-core/test/ut/request_id_policy.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+
+#include <gtest/gtest.h>
+
+using namespace Azure::Core;
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core::Http::Policies::_internal;
+
+namespace {
+class NoOpPolicy : public HttpPolicy {
```

Comment:
nit. Since we already define this TestTransportPolicy for another test, I wonder if we want some PolicyTest utilities to re-use this instead of re-defining,. 
I don't care too much since it is test, so, optional to you

Reply:
I am not against it, but let's do it as part of another PR. What I think would be the most useful is to have a transport policy that takes a lambda in the constructor, and invokes that lambda from `Send()`, with all the parameters. And the default constructor would initialize it with lambda that does not touch any parameters and returns `nullptr` response.

Reply:
#2027